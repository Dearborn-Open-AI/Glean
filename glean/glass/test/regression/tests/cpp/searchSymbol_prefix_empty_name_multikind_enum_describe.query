action: searchSymbol
args: {"name": "", "repo_name": "test", "language": [1], "kinds": [12], "options": {"detailedResults": true, "exactMatch": false, "ignoreCase": false, "namespaceSearch": false } }
