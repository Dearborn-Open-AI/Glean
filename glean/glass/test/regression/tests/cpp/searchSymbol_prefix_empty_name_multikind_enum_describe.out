[
    "@generated",
    {
        "symbolDetails": [
            {
                "comments": [],
                "kind": 12,
                "language": 1,
                "location": {
                    "filepath": "test.cpp",
                    "range": {
                        "columnBegin": 1,
                        "columnEnd": 27,
                        "lineBegin": 81,
                        "lineEnd": 81
                    },
                    "repository": "test"
                },
                "name": {
                    "container": "",
                    "localName": "Global"
                },
                "repo_hash": "testhash",
                "sym": "test/cpp//Global",
                "sym_location": {
                    "filepath": "test.cpp",
                    "range": {
                        "columnBegin": 1,
                        "columnEnd": 27,
                        "lineBegin": 81,
                        "lineEnd": 81
                    },
                    "repository": "test"
                },
                "sym_other_locations": []
            },
            {
                "comments": [],
                "kind": 12,
                "language": 1,
                "location": {
                    "filepath": "test.cpp",
                    "range": {
                        "columnBegin": 1,
                        "columnEnd": 41,
                        "lineBegin": 82,
                        "lineEnd": 82
                    },
                    "repository": "test"
                },
                "name": {
                    "container": "",
                    "localName": "GlobalClass"
                },
                "repo_hash": "testhash",
                "sym": "test/cpp//GlobalClass",
                "sym_location": {
                    "filepath": "test.cpp",
                    "range": {
                        "columnBegin": 1,
                        "columnEnd": 41,
                        "lineBegin": 82,
                        "lineEnd": 82
                    },
                    "repository": "test"
                },
                "sym_other_locations": []
            },
            {
                "comments": [],
                "kind": 12,
                "language": 1,
                "location": {
                    "filepath": "test.cpp",
                    "range": {
                        "columnBegin": 3,
                        "columnEnd": 26,
                        "lineBegin": 75,
                        "lineEnd": 75
                    },
                    "repository": "test"
                },
                "name": {
                    "container": "baz::U",
                    "localName": "InU"
                },
                "repo_hash": "testhash",
                "sym": "test/cpp//baz/U/InU",
                "sym_location": {
                    "filepath": "test.cpp",
                    "range": {
                        "columnBegin": 3,
                        "columnEnd": 26,
                        "lineBegin": 75,
                        "lineEnd": 75
                    },
                    "repository": "test"
                },
                "sym_other_locations": []
            },
            {
                "comments": [],
                "kind": 12,
                "language": 1,
                "location": {
                    "filepath": "test.cpp",
                    "range": {
                        "columnBegin": 3,
                        "columnEnd": 40,
                        "lineBegin": 76,
                        "lineEnd": 76
                    },
                    "repository": "test"
                },
                "name": {
                    "container": "baz::U",
                    "localName": "InUClass"
                },
                "repo_hash": "testhash",
                "sym": "test/cpp//baz/U/InUClass",
                "sym_location": {
                    "filepath": "test.cpp",
                    "range": {
                        "columnBegin": 3,
                        "columnEnd": 40,
                        "lineBegin": 76,
                        "lineEnd": 76
                    },
                    "repository": "test"
                },
                "sym_other_locations": []
            },
            {
                "comments": [],
                "kind": 12,
                "language": 1,
                "location": {
                    "filepath": "test.cpp",
                    "range": {
                        "columnBegin": 3,
                        "columnEnd": 32,
                        "lineBegin": 59,
                        "lineEnd": 59
                    },
                    "repository": "test"
                },
                "name": {
                    "container": "maybe::Nothing",
                    "localName": "_secret"
                },
                "repo_hash": "testhash",
                "sym": "test/cpp//maybe/Nothing/_secret",
                "sym_location": {
                    "filepath": "test.cpp",
                    "range": {
                        "columnBegin": 3,
                        "columnEnd": 32,
                        "lineBegin": 59,
                        "lineEnd": 59
                    },
                    "repository": "test"
                },
                "sym_other_locations": []
            }
        ],
        "symbols": [
            {
                "kind": 12,
                "language": 1,
                "location": {
                    "filepath": "test.cpp",
                    "range": {
                        "columnBegin": 1,
                        "columnEnd": 27,
                        "lineBegin": 81,
                        "lineEnd": 81
                    },
                    "repository": "test"
                },
                "name": "Global",
                "symbol": "test/cpp//Global"
            },
            {
                "kind": 12,
                "language": 1,
                "location": {
                    "filepath": "test.cpp",
                    "range": {
                        "columnBegin": 1,
                        "columnEnd": 41,
                        "lineBegin": 82,
                        "lineEnd": 82
                    },
                    "repository": "test"
                },
                "name": "GlobalClass",
                "symbol": "test/cpp//GlobalClass"
            },
            {
                "kind": 12,
                "language": 1,
                "location": {
                    "filepath": "test.cpp",
                    "range": {
                        "columnBegin": 3,
                        "columnEnd": 26,
                        "lineBegin": 75,
                        "lineEnd": 75
                    },
                    "repository": "test"
                },
                "name": "InU",
                "symbol": "test/cpp//baz/U/InU"
            },
            {
                "kind": 12,
                "language": 1,
                "location": {
                    "filepath": "test.cpp",
                    "range": {
                        "columnBegin": 3,
                        "columnEnd": 40,
                        "lineBegin": 76,
                        "lineEnd": 76
                    },
                    "repository": "test"
                },
                "name": "InUClass",
                "symbol": "test/cpp//baz/U/InUClass"
            },
            {
                "kind": 12,
                "language": 1,
                "location": {
                    "filepath": "test.cpp",
                    "range": {
                        "columnBegin": 3,
                        "columnEnd": 32,
                        "lineBegin": 59,
                        "lineEnd": 59
                    },
                    "repository": "test"
                },
                "name": "_secret",
                "symbol": "test/cpp//maybe/Nothing/_secret"
            }
        ]
    }
]